### PR TITLE
Use comparator for Reconfigurator

### PR DESCRIPTION
**Pull request by ywelsch**
Simplifies the voting configuration reconfiguration logic by switching to an explicit `Comparator` for the priorities. Does not make changes to the behavior of the component.

### CODE REVIEW

**Review by DaveCTurner — CHANGES_REQUESTED** [review]
The change looks good, but I would like it to be more obvious that we never add a retired node back to the config. Can we exclude such nodes from `orderedMasterNodes` entirely?

**Comment by ywelsch** [comment]
> Can we exclude such nodes from orderedMasterNodes entirely?

I think so, see 6f1c81d

File: server/src/main/java/org/elasticsearch/cluster/coordination/Reconfigurator.java [hunk 1]
```diff
@@ -102,81 +97,86 @@ public VotingConfiguration reconfigure(Set<DiscoveryNode> liveNodes, Set<String>
...
-        } else {
-            nonRetiredInConfigLiveNotMasterIds = nonRetiredInConfigLiveIds;
-            nonRetiredInConfigLiveMasterIds = Collections.emptySet();
-        }
-
-        final SortedSet<String> nonRetiredLiveNotInConfigIds = Sets.sortedDifference(liveNodeIds, currentConfig.getNodeIds());
-        nonRetiredLiveNotInConfigIds.removeAll(retiredNodeIds);
+        final Set<String> currentConfigNodeIds = currentConfig.getNodeIds();
+
+        final Set<VotingConfigNode> nonRetiredOrderedMasterNodes = new TreeSet<>();
```
Comment (DaveCTurner) [inline server/src/main/java/org/elasticsearch/cluster/coordination/Reconfigurator.java]: maybe `orderedCandidateNodes`?

**Review by DaveCTurner — APPROVED** [review]
LGTM (one optional renaming suggestion)

**Pull request merged**